pkg/subctl/cmd/join.go

```diff
@@ -378,6 +383,22 @@ func determineServiceCIDR(nd *network.ClusterNetwork) error {
 	return nil
 }
 
+// determineExternalCIDR validates and appends the external cidr to cluster cidr
+func determineExternalCIDR(nd *network.ClusterNetwork) error {
+	if externalCIDR != "" {
+		err := utils.IsValidCIDR(externalCIDR)
+		if err == nil {
+			externalCIDR = utils.GetNetworkAddress(externalCIDR)
+			nd.PodCIDRs = append(nd.PodCIDRs, externalCIDR)
+			clusterCIDR = strings.Join(nd.PodCIDRs, ",")
+		} else {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
We tend to structure code with early returns to reduce nesting and I think this style is more or less idiomatic in Go.

```suggestion
	if externalCIDR == "" {
	    return nil
	}
	
        err := utils.IsValidCIDR(externalCIDR)
	if err != nil {
	    return err
	}
		
        externalCIDR = utils.GetNetworkAddress(externalCIDR)
        nd.PodCIDRs = append(nd.PodCIDRs, externalCIDR)
        clusterCIDR = strings.Join(nd.PodCIDRs, ",")

	return nil
```

Reply:
ahhh, I thought it was vice versa. will move it over to the `cmd/subctl` package. ty for the info.